hera_sim/tests/test_adjustment.py

```diff
@@ -0,0 +1,695 @@
+"""Test the various simulation adjustment tools."""
+
+import copy
+import itertools
+import os
+import pytest
+
+from astropy import units
+from scipy import stats
+import numpy as np
+
+from hera_sim import adjustment
+from hera_sim import antpos
+from hera_sim import interpolators
+from hera_sim import Simulator
+from pyuvdata import UVData
+
+# For use later when simulating foregrounds.
+Tsky_mdl = interpolators.Tsky(datafile="HERA_Tsky_Reformatted.npz")
+omega_p = interpolators.Beam(datafile="HERA_H1C_BEAM_POLY.npy", interpolator="poly1d")
+
+
+@pytest.fixture(scope="session")
+def base_config():
+    Nfreqs = 100
+    start_freq = 100e6
+    bandwidth = 50e6
+    Ntimes = 100
+    start_time = 2458799.0
+    integration_time = 10.7
+    # Use square array with HERA antenna separation.
+    base_sep = 14.6
+    array_layout = {
+        0: [0, 0, 0],
+        1: [1, 0, 0],
+        2: [0, 1, 0],
+        3: [1, 1, 0],
+    }
+    array_layout = {ant: base_sep * np.array(pos) for ant, pos in array_layout.items()}
+    config = dict(
+        Nfreqs=Nfreqs,
+        start_freq=start_freq,
+        bandwidth=bandwidth,
+        Ntimes=Ntimes,
+        start_time=start_time,
+        integration_time=integration_time,
+        array_layout=array_layout,
+    )
+    return config
+
+
+def antpos_equal(antpos_1, antpos_2, tol=0.1):
+    return all(
+        any(np.allclose(pos_1, pos_2, atol=tol) for pos_1 in antpos_1.values())
+        for pos_2 in antpos_2.values()
+    )
+
+
+def test_basic_antenna_matching():
+    # Simple test: reflected right-triangles should match exactly
+    array_1 = {0: [0, 0, 0], 1: [1, 0, 0], 2: [1, 1, 0]}
+
+    array_2 = {0: [0, 0, 0], 1: [0, 1, 0], 2: [1, 1, 0]}
+
+    array_intersection = adjustment._get_array_intersection(array_1, array_2, tol=0)[0]
+    assert antpos_equal(array_intersection, array_2, 0)
+
+    # In general, rotated right-triangles should have at least two antennas in their
+    # intersection.
+    array_3 = {0: [0, 0, 0], 1: [1, 0, 0], 2: [0, 1, 0]}
+    array_intersection = adjustment._get_array_intersection(array_1, array_3, tol=0)[0]
+    assert len(array_intersection) == 2
+
+    # A simple translation should just be undone
+    translation = np.random.uniform(-1, 1, 3)
+    array_4 = {ant: np.array(pos) - translation for ant, pos in array_1.items()}
+    array_intersection = adjustment._get_array_intersection(array_1, array_4, tol=0.1)[
+        0
+    ]
+    assert antpos_equal(array_intersection, array_4, 0.1)
+
+    # A small hex array should be a subset of a larger hex array
+    hex_array = antpos.HexArray(split_core=False, outriggers=0)
+    hex_array_1 = hex_array(3)
+    hex_array_2 = hex_array(2)
+    array_intersection = adjustment._get_array_intersection(hex_array_1, hex_array_2)[0]
+    assert antpos_equal(array_intersection, hex_array_2)
+    assert len(array_intersection) == len(hex_array_2)
+
+
+def test_match_antennas(base_config):
```

Comment:
this test is enormous and tests a bunch of different things. You should look into using `pytest.fixture`s and parameterization to break it into a bunch of smaller tests, if possible.

Reply:
Gotcha, I'll look into breaking it up. I agree that it's a huge test and could be compartmentalized.